src/main/java/adris/altoclef/tasks/misc/GetToOuterEndIslandsTask.java

```diff
@@ -0,0 +1,120 @@
+package adris.altoclef.tasks.misc;
+
+import adris.altoclef.AltoClef;
+import adris.altoclef.Debug;
+import adris.altoclef.tasks.InteractWithBlockTask;
+import adris.altoclef.tasks.movement.GetToYTask;
+import adris.altoclef.tasks.resources.GetBuildingMaterialsTask;
+import adris.altoclef.tasks.speedrun.BeatMinecraft2Task;
+import adris.altoclef.tasks.squashed.CataloguedResourceTask;
+import adris.altoclef.tasksystem.Task;
+import adris.altoclef.util.Dimension;
+import adris.altoclef.util.ItemTarget;
+import adris.altoclef.util.baritone.GoalAnd;
+import adris.altoclef.util.helpers.StorageHelper;
+import adris.altoclef.util.helpers.WorldHelper;
+
+import baritone.api.pathing.goals.GoalComposite;
+import baritone.api.pathing.goals.GoalGetToBlock;
+import baritone.api.pathing.goals.GoalYLevel;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.Items;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.math.Vec3i;
+
+public class GetToOuterEndIslandsTask extends Task {
+    private Task _beatTheGame;
+    private boolean _pathingToGateway = false;
+    public final Vec3i[] OFFSETS = {
+            new Vec3i(1, -1, 1),
+            new Vec3i(1, -1, -1),
+            new Vec3i(-1, -1, 1),
+            new Vec3i(-1, -1, -1),
+            new Vec3i(2, -1, 0),
+            new Vec3i(0, -1, 2),
+            new Vec3i(-2, -1, 0),
+            new Vec3i(0, -1, -2)
+    };
+    public GetToOuterEndIslandsTask() {
+
+    }
+
+    @Override
+    protected void onStart(AltoClef mod) {
+        mod.getBehaviour().push();
+        mod.getBlockTracker().trackBlock(Blocks.END_GATEWAY);
+        _beatTheGame = new BeatMinecraft2Task();
+    }
+
+    @Override
+    protected Task onTick(AltoClef mod) {
+        if (isFinished(mod)) {
+            setDebugState("What?");
+            return null;
+        }
+        if (mod.getBlockTracker().anyFound(Blocks.END_GATEWAY)) {
+            if (!mod.getItemStorage().hasItemInventoryOnly(Items.ENDER_PEARL)) {
+                setDebugState("Getting an ender pearl");
+                return new CataloguedResourceTask(new ItemTarget(Items.ENDER_PEARL, 1));
+            }
+            BlockPos gateway = mod.getBlockTracker().getNearestTracking(Blocks.END_GATEWAY).get();
+            int blocksNeeded = Math.abs(mod.getPlayer().getBlockY() - gateway.getY()) +
+                    Math.abs(mod.getPlayer().getBlockX() - gateway.getX()) +
+                    Math.abs(mod.getPlayer().getBlockZ() - gateway.getZ()) - 1;
+            if (StorageHelper.getBuildingMaterialCount(mod) < blocksNeeded) {
+                setDebugState("Getting building materials");
+                return new GetBuildingMaterialsTask(blocksNeeded);
+            }
+            GoalAnd goal = makeGoal(gateway);
+            Debug.logMessage(mod.getPlayer().getBlockPos().toString());
+            if (!goal.isInGoal(mod.getPlayer().getBlockPos()) || !mod.getPlayer().isOnGround()) {
```

Comment:
Out of curiosity, why re-invent the wheel? You could use Altoclef's built in gotoXZ and gotoY functions for this. I would change this.

Reply:
I dunno I thought it would be fun and a bit more efficient since I'm allowing baritone to approach from any side

Reply:
Also its fun to reinvent the wheel

Reply:
agreed, it can be fun as heck